sklearn/ensemble/gradient_boosting.py

```diff
@@ -945,6 +945,11 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             classification, splits are also ignored if they would result in any
             single class carrying a negative weight in either child node.
 
+        X_idx_sorted : array-like, shape = [n_samples, n_features], optional
+            The indexes of the sorted training input samples. If many tree
```

Comment:
It's unclear to me whether `X_idx_sorted[i]` is the rank of `X[i]` in each column or whether `X_idx_sorted[i, j]` indicates the index of the ith ranked value of feature j.

Better off just providing code here for how to generate `X_idx_sorted`.